armada/handlers/tiller.py

```diff
@@ -134,18 +133,138 @@ def list_charts(self):
+                                       .format(pod.metadata.name))
+                elif pod.status.phase == TERMINATED_STATUS:
+                    continue
+                elif pod.metadata.name == deleted and not (deleted == ''):
+                    continue
+                elif (pod.status.phase == RUN_STATUS or
+                      pod.status.phase == SUCCESS_STATUS):
+                    pods.remove(pod)
+
+            if not (len(pods) == 0):
+                pods = self.k8s.get_namespace_pod(
+                    namespace,
+                    label=labels).items
+
+    def update_statefull_sets(self, name, release_name, namespace, labels,
+                              action_type, chart, disable_hooks, values):
+        '''
+        update statefullsets (daemon, stateful)
+        '''
+
+        if action_type == 'daemonset':
+
+            LOG.info('Updating: %s', action_type)
+            label_selector = 'release_name={}'.format(release_name)
+
+            if not (labels is None):
+                for label in labels:
+                    label_selector += ', {}={}'.format(label.keys()[0],
+                                                       label.values()[0])
+
+            get_daemonset = self.k8s.get_namespace_daemonset(
+                namespace=namespace, label=label_selector)
+
+            for ds in get_daemonset.items:
+                ds_name = ds.metadata.name
+                ds_labels = ds.metadata.labels
+                if ds_name == name:
+                    LOG.info("Deleting %s : %s in %s", action_type, ds_name,
+                             namespace)
+                    self.k8s.delete_daemon_action(ds_name, namespace)
+
+                    # update the daemonset yaml
+                    template = self.get_chart_templates(
+                        ds_name, name, release_name, namespace, chart,
+                        disable_hooks, values)
+                    template['metadata']['labels'] = ds_labels
+                    template['spec']['template']['metadata']['labels'] = ds_labels
+                    self.k8s.create_daemon_action(
+                        namespace=namespace, template=template)
+
+                    # delete pods
+                    pods = self.k8s.get_namespace_pod(
+                        namespace=namespace, label=label_selector)
+
+                    for pod in pods.items:
+                        pod_name = pod.metadata.name
+
+                        if name in pod_name:
+                            LOG.info('Deleting %s pod in %s', pod_name,
+                                     namespace)
+
+                            self.k8s.delete_namespace_pod(
+                                pod_name, namespace=namespace)
+                            # wait for pod to come back up
+
+                            self.wait_for_deployment(namespace,
+                                                     label_selector, pod_name)
+
+        elif action_type == 'statefulset':
```

Comment:
the name should not matter if it is daemonset or statefulset. i think it would be appropriate for both. unless i am not thinking of a scenario